AVS 2023-03-01 SDK generation

nivasn-msft opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

nivasn-msft committed:
AVS 2023-03-01 SDK generation

nivasn-msft committed:
AVS 2023-03-01 SDK generation

nivasn-msft committed:
Updating version

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

live1206 commented on sdk/avs/Azure.ResourceManager.Avs/CHANGELOG.md:
```diff
@@ -1,5 +1,15 @@
+### Other Changes
+- Upgraded API version to 2023-03-01
+
+
```
Please remove the newly added entry, update the existing entry for `1.2.0-beta.2` instead.

nivasn-msft replied:
updated.

nivasn-msft committed:
updating version

jsquire commented:
@nivasn-msft:

Our automation is identifying you as a community contributor.  Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

nivasn-msft commented:
/azp run

nivasn-msft committed:
updating code owners

nivasn-msft commented:
Hello Jesse,

I have added myself to the azure-sdk-read github team (https://github.com/orgs/Azure/teams/azure-sdk-read?query= ) , and have also added myself to the CODEOWNERS<https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS> . My PR (https://github.com/Azure/azure-sdk-for-net/pull/39536) still continues to show community contributor tag. Is there anything else I am supposed to do ?

Thanks
Nivas

From: Jesse Squire ***@***.***>
Sent: Wednesday, November 1, 2023 9:27 AM
To: Azure/azure-sdk-for-net ***@***.***>
Cc: Nivas Narayanasamy ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/azure-sdk-for-net] AVS 2023-03-01 SDK generation (PR #39536)


@nivasn-msft<https://github.com/nivasn-msft>:

Our automation is identifying you as a community contributor. Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct.

  *   Azure SDK onboarding<https://eng.ms/docs/products/azure-developer-experience/onboard/requirements> (Microsoft internal)
  *   Azure SDK onboarding assistance<https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47> (Microsoft internal)

Please also be sure to add yourself to CODEOWNERS<https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS> for this library, if you will be maintaining it going forward.

-
Reply to this email directly, view it on GitHub<https://github.com/Azure/azure-sdk-for-net/pull/39536#issuecomment-1788955538>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2OYNQXQMGJSN7Y6CXFJD5DYCJE2RAVCNFSM6AAAAAA6R3NGIKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBYHE2TKNJTHA>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

jsquire commented:
> Hello Jesse, I have added myself to the azure-sdk-read github team (https://github.com/orgs/Azure/teams/azure-sdk-read?query= ) , and have also added myself to the CODEOWNERS<https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS> . My PR (#39536) still continues to show community contributor tag. Is there anything else I am supposed to do ? 

You'll need write permissions to the repository in order to be a contributor and valid entry in CODEOWNERS.

nivasn-msft committed:
adding test cases and assets

nivasn-msft committed:
Merge branch 'main' of https://github.com/nivasn-msft/azure-sdk-for-net into nivasn/avs_2023-03-01_sdk

nivasn-msft committed:
updating changelog

nivasn-msft committed:
test cases for placement policy

cataggar commented on sdk/avs/Azure.ResourceManager.Avs/api/Azure.ResourceManager.Avs.netstandard2.0.cs:
```diff
@@ -1123,7 +1124,7 @@ public static partial class ArmAvsModelFactory
         public static Azure.ResourceManager.Avs.AvsPrivateCloudClusterData AvsPrivateCloudClusterData(Azure.Core.ResourceIdentifier id = null, string name = null, Azure.Core.ResourceType resourceType = default(Azure.Core.ResourceType), Azure.ResourceManager.Models.SystemData systemData = null, string skuName = null, int? clusterSize = default(int?), Azure.ResourceManager.Avs.Models.AvsPrivateClou ...
         public static Azure.ResourceManager.Avs.AvsPrivateCloudClusterVirtualMachineData AvsPrivateCloudClusterVirtualMachineData(Azure.Core.ResourceIdentifier id = null, string name = null, Azure.Core.ResourceType resourceType = default(Azure.Core.ResourceType), Azure.ResourceManager.Models.SystemData systemData = null, string displayName = null, string moRefId = null, string folderPath = null,  ...
-        public static Azure.ResourceManager.Avs.AvsPrivateCloudData AvsPrivateCloudData(Azure.Core.ResourceIdentifier id = null, string name = null, Azure.Core.ResourceType resourceType = default(Azure.Core.ResourceType), Azure.ResourceManager.Models.SystemData systemData = null, System.Collections.Generic.IDictionary<string, string> tags = null, Azure.Core.AzureLocation location = default(Azure. ...
+        public static Azure.ResourceManager.Avs.AvsPrivateCloudData AvsPrivateCloudData(Azure.Core.ResourceIdentifier id = null, string name = null, Azure.Core.ResourceType resourceType = default(Azure.Core.ResourceType), Azure.ResourceManager.Models.SystemData systemData = null, System.Collections.Generic.IDictionary<string, string> tags = null, Azure.Core.AzureLocation location = default(Azure. ...
```
This adds `System.Collections.Generic.IEnumerable<string> extendedNetworkBlocks = null,`, not sure my CI is complaining about API compatibility. https://github.com/Azure/azure-sdk-for-net/pull/39536/checks?check_run_id=19520494032

cataggar replied:
> Error : MembersMustExist : Member 'public Azure.ResourceManager.Avs.AvsPrivateCloudData Azure.ResourceManager.Avs.Models.ArmAvsModelFactory.AvsPrivateCloudData(Azure.Core.ResourceIdentifier, System.String, Azure.Core.ResourceType, Azure.ResourceManager.Models.SystemData, System.Collections.Generic.IDictionary<System.String, System.String>, Azure.Core.AzureLocation, System.String, Azure.ResourceManager.Models.ManagedServiceIdentity, Azure.ResourceManager.Avs.Models.AvsManagementCluster, System.Nullable<Azure.ResourceManager.Avs.Models.InternetConnectivityState>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.Avs.Models.SingleSignOnIdentitySource>, Azure.ResourceManager.Avs.Models.PrivateCloudAvailabilityProperties, Azure.ResourceManager.Avs.Models.CustomerManagedEncryption, System.Nullable<Azure.ResourceManager.Avs.Models.AvsPrivateCloudProvisioningState>, Azure.ResourceManager.Avs.Models.ExpressRouteCircuit, Azure.ResourceManager.Avs.Models.AvsPrivateCloudEndpoints, System.String, System.String, System.String, System.String, System.String, System.String, System.String, System.String, System.Collections.Generic.IEnumerable<Azure.Core.ResourceIdentifier>, Azure.ResourceManager.Avs.Models.ExpressRouteCircuit, System.Nullable<Azure.ResourceManager.Avs.Models.NsxPublicIPQuotaRaisedEnum>)' does not exist in the implementation but it does exist in the contract.

nivasn-msft replied:
Was able to add the old `AvsPrivateCloudData  ` method to `customized/ArmAvsModelFactory.cs public static partial class ArmAvsModelFactor`

jsquire commented:
@nivasn-msft:  Your Microsoft and Azure organization memberships need to be made public.

nivasn-msft committed:
avs private cloud patch

nivasn-msft committed:
adding placement policy test and public partial class avs model factory

nivasn-msft commented:
> @nivasn-msft: Your Microsoft and Azure organization memberships need to be made public. Also, the labels are added by automation when the PR is opened, they're not reevaluated.

I have made my membership public

jsquire commented:
> I have made my membership(s) public

Confirmed.  Permissions and memberships look good!

nivasn-msft committed:
running export-api script

nivasn-msft committed:
Merge branch 'main' of https://github.com/nivasn-msft/azure-sdk-for-net into nivasn/avs_2023-03-01_sdk

nivasn-msft commented:
@live1206 @jsquire ,could you please review the PR when you get a chance.

live1206 commented on sdk/avs/Azure.ResourceManager.Avs/CHANGELOG.md:
```diff
@@ -1,14 +1,12 @@
 # Release History
 
-## 1.3.0-beta.1 (Unreleased)
+## 1.3.0-beta.1 (2023-12-11)
```
This date is the release date of .NET SDK, which should be a future date.

nivasn-msft replied:
updated.

live1206 commented on sdk/avs/Azure.ResourceManager.Avs/test-resources.json:
```diff
@@ -0,0 +1,6 @@
+{
```
it seems this is not in use?

nivasn-msft replied:
yes, would like to have it as a placeholder ,if in future we are deploying test resources using arm template

live1206 replied:
May I know what test resources have to be created manually beforehand?
Basically, the test resources are supposed to be created using .NET SDK in the test or during the test setup, so that it's easier to re-record the tests by anyone.
For instance, test resources are created with random name during test setup in [this](https://github.com/Azure/azure-sdk-for-net/blob/9ac52a165101b727ff606dd50dd82f771ec7c710/sdk/sqlvirtualmachine/Azure.ResourceManager.SqlVirtualMachine/tests/Scenario/SqlVmGroupTests.cs#L28)

live1206 replied:
OK then

live1206 commented on sdk/avs/Azure.ResourceManager.Avs/tests/AvsManagementTestBase.cs:
```diff
@@ -37,5 +47,29 @@ protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionReso
+        protected async Task<AvsPrivateCloudResource> getAvsPrivateCloudResource()
+        {
+            string subscriptionId = DefaultSubscription.Data.SubscriptionId;
+            string resourceGroupName = "avs-sdk-test";
```
why do we have the resource group name hard-coded? is it because it was created beforehand?

nivasn-msft replied:
Yes, it was created before hand and locked for sdk testing.

live1206 commented on sdk/avs/Azure.ResourceManager.Avs/tests/AvsManagementTestBase.cs:
```diff
@@ -37,5 +47,29 @@ protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionReso
+        protected async Task<AvsPrivateCloudClusterResource> getAvsPrivateCloudClusterResource()
+        {
+            string subscriptionId = DefaultSubscription.Data.SubscriptionId;
+            string resourceGroupName = "avs-sdk-test";
```
if we really want to have this hard-coded, we can put this as const class fields.

nivasn-msft replied:
done.

nivasn-msft committed:
PR comments fixes

nivasn-msft committed:
Merge branch 'Azure:main' into nivasn/avs_2023-03-01_sdk

nivasn-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into nivasn/avs_2023-03-01_sdk

nivasn-msft committed:
Merge branch 'nivasn/avs_2023-03-01_sdk' of https://github.com/nivasn-msft/azure-sdk-for-net into nivasn/avs_2023-03-01_sdk

nivasn-msft committed:
running generate code
